### PR TITLE
tests: Increase coverage for msg.go and access_list_tx.go

### PR DESCRIPTION
**Pull request by MalteHerrmann**
Addresses: #1017

## Description

- Added tests for `msg.go` and `access_list_tx.go`
- Replaced deprecated `ethtypes.NewTransaction` with `ethtypes.NewTx` in some tests (https://pkg.go.dev/github.com/ethereum/go-ethereum/core/types#NewTransaction)

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/types/access_list_tx_test.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/suite"
+	"github.com/tharsis/ethermint/tests"
+)
+
+type AccessListTxTestSuite struct {
```
Comment (fedekunze) [inline x/evm/types/access_list_tx_test.go:14]: nit, I think we could use the same test suite as the other transaction types

File: x/evm/types/access_list_tx_test.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+func TestAccessListTxTestSuite(t *testing.T) {
+	suite.Run(t, new(AccessListTxTestSuite))
+}
+
+func (suite *AccessListTxTestSuite) TestAccessListTxCopy() {
+	tx := &AccessListTx{}
+	txCopy := tx.Copy()
+
+	suite.Require().Equal(&AccessListTx{}, txCopy)
+	// TODO: Test for different pointers
```
Comment (fedekunze) [inline x/evm/types/access_list_tx_test.go:49]: address issue?

File: x/evm/types/msg_test.go [hunk 3]
```diff
@@ -70,73 +73,93 @@ func (suite *MsgsTestSuite) TestMsgEthereumTx_Constructor() {
 }
 
 func (suite *MsgsTestSuite) TestMsgEthereumTx_BuildTx() {
-	msg := types.NewTx(nil, 0, &suite.to, nil, 100000, big.NewInt(1), big.NewInt(1), big.NewInt(0), []byte("test"), nil)
+	testCases := []struct {
```
Comment (fedekunze) [inline x/evm/types/msg_test.go:76]: nice refactor!

File: x/evm/types/msg_test.go [hunk 4]
```diff
@@ -70,73 +73,93 @@ func (suite *MsgsTestSuite) TestMsgEthereumTx_Constructor() {
...
+		{msg: "negative gas price - AccessListTx", to: suite.to.Hex(), amount: hundredInt, gasPrice: minusOneInt, gasFeeCap: nil, gasTipCap: nil, accessList: &ethtypes.AccessList{}, chainID: nil, expectPass: false},
+		{msg: "zero gas price - AccessListTx", to: suite.to.Hex(), amount: hundredInt, gasPrice: zeroInt, gasFeeCap: nil, gasTipCap: nil, accessList: &ethtypes.AccessList{}, chainID: hundredInt, expectPass: true},
+		{msg: "invalid from address - AccessListTx", to: suite.to.Hex(), amount: hundredInt, gasPrice: zeroInt, gasFeeCap: nil, gasTipCap: nil, from: invalidFromAddress, accessList: &ethtypes.AccessList{}, chainID: hundredInt, expectPass: false},
+		{msg: "chain ID not set on AccessListTx", to: suite.to.Hex(), amount: hundredInt, gasPrice: zeroInt, gasFeeCap: nil, gasTipCap: nil, accessList: &ethtypes.AccessList{}, chainID: nil, expectPass: false},
+		{msg: "nil tx.Data - AccessList Tx", to: suite.to.Hex(), amount: hundredInt, gasPrice: zeroInt, gasFeeCap: nil, gasTipCap: nil, accessList: &ethtypes.AccessList{}, chainID: hundredInt, expectPass: false},
```
Comment (fedekunze) [inline x/evm/types/msg_test.go:149]: I'd probably split this into multiple lines to ease readibility

**Review by fedekunze — COMMENTED** [review]
LGTM. minor comments

File: x/evm/types/access_list_tx_test.go (hunk 2 above)
Comment (MalteHerrmann) [reply x/evm/types/access_list_tx_test.go:49]: TBH I don't know the necessity for this - this comment exists both in `TestLegacyTxCopy` as well as `TestDynamicFeeTxCopy`, so I figured this should apply here too.

File: x/evm/types/access_list_tx_test.go (hunk 1 above)
Comment (MalteHerrmann) [reply x/evm/types/access_list_tx_test.go:14]: Absolutely, I'll change it. 

Reason I added this was, that the VS Code Go extension would not let me debug the tests individually when the test suite is defined in another file.

File: x/evm/types/msg_test.go (hunk 4 above)
Comment (MalteHerrmann) [reply x/evm/types/msg_test.go:149]: I agree - I kept the original format before my changes but will gladly change it.

**Comment by MalteHerrmann** [comment]
Thanks for the comments @fedekunze, I've pushed the corresponding changes.

**Review by fedekunze — APPROVED** [review]

**Comment by loredanacirstea** [comment]
@MalteHerrmann check the Evmos Discord `contributors` channel - for external contributors to have tech discussions and you can ask for access with this PR link.

**Pull request merged**